compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12647,6 +12659,195 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
+                 op == TR::sadd || op == TR::badd || op == TR::buadd);
+
+   bool isSub = (op == TR::isub || op == TR::lsub || op == TR::iusub || op == TR::lusub ||
+                 op == TR::ssub || op == TR::bsub || op == TR::busub);
+
+   if ((isAdd || isSub) &&
+           firstChild->chkCannotOverflow()  &&
+           firstChild->getSecondChild()->getOpCode().isLoadConst() &&
+           secondChild->getOpCode().isLoadConst() &&
+           firstChild->isSingleRef() &&
+           !(!firstChild->isSingleRef() &&
+             firstChild->getFutureUseCount() != firstChild->getReferenceCount() - 1))
+      {
+      bool isUnsigned = firstChild->getOpCode().isUnsigned();
+
+      int64_t const1 = 0, const2 = 0, const3 = 0;
+
+      // Get and check the integer/long constant values
+      if (isInt)
+         {
+         const1 = firstChild->getSecondChild()->getConstValue();
+         const2 = secondChild->getConstValue();
+         const3 = isAdd ? (const2 - const1) : (const2 + const1);
+
+         int64_t signedMax = 0, signedMin = 0;
+         uint64_t unsignedMax = 0, unsignedMin = 0;
+
+         if (firstChild->getOpCode().is1Byte())
+            {
+            signedMax   = !isUnsigned ? TR::getMaxSigned<TR::Int8>()   : 0;
+            signedMin   = !isUnsigned ? TR::getMinSigned<TR::Int8>()   : 0;
+            unsignedMax = isUnsigned  ? TR::getMaxUnsigned<TR::Int8>() : 0;
+            unsignedMin = isUnsigned  ? TR::getMinUnsigned<TR::Int8>() : 0;
+            }
+         else if (firstChild->getOpCode().is2Byte())
+            {
+            signedMax   = !isUnsigned ? TR::getMaxSigned<TR::Int16>()   : 0;
+            signedMin   = !isUnsigned ? TR::getMinSigned<TR::Int16>()   : 0;
+            unsignedMax = isUnsigned  ? TR::getMaxUnsigned<TR::Int16>() : 0;
+            unsignedMin = isUnsigned  ? TR::getMinUnsigned<TR::Int16>() : 0;
+            }
+         else if (firstChild->getOpCode().is4Byte())
+            {
+            signedMax   = !isUnsigned ? TR::getMaxSigned<TR::Int32>()   : 0;
+            signedMin   = !isUnsigned ? TR::getMinSigned<TR::Int32>()   : 0;
+            unsignedMax = isUnsigned  ? TR::getMaxUnsigned<TR::Int32>() : 0;
+            unsignedMin = isUnsigned  ? TR::getMinUnsigned<TR::Int32>() : 0;
+            }
+         else
+            {
+            TR_ASSERT(0, "Unable to handle a number that's bigger than 4 bytes.");
+            }
+
+         if ((isUnsigned && (const3 > unsignedMax || const3 < unsignedMin)) ||
+                 (!isUnsigned && (const3 > signedMax || const3 < signedMin)))
+            {
+            return node;
+            }
+         }
+      else  // Long
+         {
+         if (isUnsigned)
+            {
+            uint64_t uconst1 = firstChild->getSecondChild()->getUnsignedLongInt();
+            uint64_t uconst2 = secondChild->getUnsignedLongInt();
+            uint64_t uconst3 = 0;
```

Comment:
This `uconst3` is unused after assigned, and the value of `const3` is left unchanged in this branch